packages/replay/src/util/addMemoryEntry.ts

```diff
@@ -0,0 +1,19 @@
+import { WINDOW } from '../constants';
+import type { ReplayContainer } from '../types';
+import { createPerformanceSpans } from './createPerformanceSpans';
+
+/**
+ * Create a "span" for the total amount of memory being used by JS objects
+ * (including v8 internal objects).
+ */
+export function addMemoryEntry(replay: ReplayContainer): Promise<void[]> | undefined {
```

Comment:
l: can we change the func signature here to just be void? Then we don't need the `return` statements and we can also just void `createPerformanceSpans`

Reply:
`addEvent` was also sync before, and we are not really waiting for it anywhere?

Reply:
🤔 , it may have been a bug -- eventbuffer's addevent is async, looks like replay.addEvent was not returning the promise.

Reply:
OK, I think you're right - but I'll make a separate PR to fix that 👍